Documentation/contributor-guide/roadmap.md

```diff
@@ -0,0 +1,42 @@
+| Deprecate grpc-gateway or bump it                                  | P1       | It isn't guaranteed to be resolved in 3.6, and might be postponed to 3.7 depending on the effort and risk.   |
+| Release raft 3.6.0                                                 | P1       | etcd 3.6.0 will depends on raft 3.6.0                                                                        |
+| bbolt: Add logger into bbolt                                       | P1       | It's important to diagnose bbolt issues                                                                      |
+| bbolt: Add surgery commands                                        | P1       | Surgery commands are important for fixing corrupted db files                                                 |
+| Release bbolt 1.3.8                                                | P1       | etcd 3.6.0 will depends on bbolt 1.3.8                                                                       |
+
+## v3.7.0
+
+| Title                                                             | Priority | Note                                                                              |
+|-------------------------------------------------------------------|----------|-----------------------------------------------------------------------------------|
+| StoreV2 deprecation                                               | P0       | Finish the remaining tasks 3.7.                                                   |
+| Evaluate and (graduate or deprecate/remove) experimental features | P2       | Finish the remaining tasks 3.7.                                                   |
+| Refactor lease: Lease might be revoked by mistake by old leader   | P0       | to be investigated & discussed                                                    |
+| Integrate raft's new feature (async write) into etcd              | P1       | It should can improve the performance                                             |
+| bbolt: Support customizing the bbolt rebalance threshold          | P2       | It may get rid of etcd's defragmentation. Both bbolt and etcd need to be changed. |
+
+## Backlog (future releases)
+
+| Title                                                     | Priority | Note |
+|-----------------------------------------------------------|----------|------|
+| Remove the dependency on grpc-go's experimental API       |          |      |
+| Protobuf: cleanup both golang/protobuf and gogo/protobuf  |          |      |
+| Proposals should include a merkle root                    |          |      |
+| Add Distributed Tracing using OpenTelemetry               |          |      |
+| Support CA rotation                                       |          |      |
+| bbolt: Migrate all commands to cobra style commands       |          |      |
+| raft: enhance the configuration change validation         |          |      |
```

Comment:
What's the strategy of priority changes? Is the roadmap subject to change say one story is completed and backlog item is auto poped up into the next version roadmap?

Reply:
Basically I don't think there is much change on 3.6 and 3.7 on the important items. Just as I mentioned in the roadmap, it just records the most important tasks for each major or minor releases. It doesn't prevent other small & safe changes.

Reply:
I am not saying the items in the roadmap are not set in stone. We can also update it in future based on agreement in maintainers and contributors.

Reply:
Thanks @ahrtr for clarification !